app/controllers/submissions_controller.rb

```diff
@@ -168,8 +168,17 @@ def destroy
   end
 
   # POST
-  def soft_delete
-    @submission.soft_deleted = true
+  def mod_edit
+    if params.has_key? :soft_deleted
+      @submission.soft_deleted = (params[:soft_deleted] == "true")
+    end
+    if params.has_key? :approved
+      @submission.approved = (params[:approved] == "true")
+    end
+    if params.has_key? :nsfw_level
+      @submission.nsfw_level = params[:nsfw_level]
+    end
+
     @submission.save
```

Comment:
You might want to use `.save!` for the reasons I mentioned previously. Also, prefer single quotes to double quotes. Please turn on the linter; you said you have it on but watching your string it's obvious that you don't, or at the very least it's ignoring the project settings.
Also FWIW, Ruby has a `.to_b` which behaves similarly to `== 'true'`, though I'm not sure if that means you should use it.